src/hypothesis/strategies.py

```diff
@@ -1681,7 +1681,7 @@ def try_convert(typ, value, name):
                 name, value, type(value).__name__, typ.__name__
             )
         )
-    except ValueError:
+    except (OverflowError, ValueError):
```

Comment:
What caused this to become necessary?

Reply:
Somewhere in Pandas index handling, from vague memory in uint64 or datetime64, casting Python integers to the specific type raises OverflowError.  This seemed like the best place to handle it, and already has the right error message.